frontend/src/components/GameBoard/GameBoard.tsx

```diff
@@ -34,10 +38,43 @@ const GameBoard = observer((): ReactElement => {
   }, []);
 
   useEffect(() => {
-    onTilePlacement(0, 0);
+    onTilePlacement(0, 0, false);
     rootStore.gameStore.endCurrentTurn();
   }, []);
 
+  useEffect(() => {
+    socket.on(WebSocketEvent.RECEIVE_TILE_PLACED, (data) => {
+      const { tileData, clientId } = data;
+      const { id, row, column, rotation } = websocketMessageParser.parse(tileData, WebSocketEvent.RECEIVE_TILE_PLACED);
+
+      if (row !== 0 || column !== 0) {
+        rootStore.gameStore.setTileInHandFromWebSocket(id, rotation);
+        onTilePlacement(row, column, true);
+      }
+
+      console.log(
+        `Tile with id ${id} rotated ${rotation} degrees is placed in ${row} row and ${column} column by client with id ${clientId}`,
+      );
+    });
+  }, []);
+
+  useEffect(() => {
+    socket.on(WebSocketEvent.RECEIVE_TILE_ROTATED, (data) => {
+      const { rotation, clientId } = data;
+      const rotationDegree = rotation as Rotation;
+      rootStore.gameStore.setRotationFromWebSocket(rotationDegree);
+      console.log(`Client with id ${clientId} rotated tile in hand ${rotationDegree} degrees`);
+    });
+  }, []);
+
+  useEffect(() => {
+    socket.on(WebSocketEvent.RECEIVE_NEXT_PHASE, (data) => {
+      const { nextPhase, clientId } = data;
+      nextPhase && rootStore.gameStore.setNextPhase(true);
+      console.log(`Client with id ${clientId} moved to the next phase.`);
+    });
+  }, []);
+
```

Comment:
please extract it to the custom hook (eg. useTileWebsocket). This can be placed in components/GameBoard/GameBoard.hooks.ts file

Reply:
done